internal/pkg/configloader/config.go

```diff
@@ -19,6 +21,19 @@ var (
 // Config is the struct for loading DevStream configuration YAML files.
 type Config struct {
 	Tools []Tool `yaml:"tools"`
+	State *State
+}
+
+// GeneralConfig is the struct for loading State and configuration YAML files.
+type GeneralConfig struct {
```

Comment:
I suggest changing "GeneralConfig" to "ConfigFile" (only an example; the idea is to relate it to the actual config.) 

Otherwise, "Config" and "GeneralConfig" are very similar; developers can't tell their names and usage of them at a glance. "Config" is what the "LoadConf" function returns, and "GeneralConfig" is actually the config.yaml file.

Other comments need to be updated as well related to this.

Reply:
+1 from me.
GeneralConfig has no usage here.